viewer/viewer.js

```diff
@@ -4514,6 +4517,136 @@ app.get('/:nodeName/:id/bodypng/:bodyType/:bodyNum/:bodyName', checkProxyRequest
   });
 });
 
+/**
+ * Get a file given a hash of that file
+ */
+
+app.get('/huntfile/hash/:hash', logAction('huntfile'), function(req, res) {
+
+  var hash = null;
+  var nodeName = null;
+  var sessionID = null;
+
+  buildSessionQuery(req, function(bsqErr, query, indices) {
+    if (bsqErr) {
+      res.status(400);
+      return res.end(bsqErr);
+    }
+    if (Config.debug) {
+      console.log(`sessions.json ${indices} query`, JSON.stringify(query, null, 1));
+    }
+
+    Db.searchPrimary(indices, 'session', query, function(err, sessions) {
+      if (err ) {
+        console.log ("Error -> Db Search ", err);
+        res.status(400);
+        res.end(err);
+      } else if (sessions.error) {
+        console.log ("Error -> Db Search ", sessions.error);
+        res.status(400);
+        res.end(sessions.error);
+      } else {
```

Comment:
General style would be no else, and to add returns to the ifs, just so there isn't as much indentation.